welkin/models/base.py

```diff
@@ -20,11 +23,23 @@ def __getattr__(self, name):
 
 
 class Resource(dict, SchemaBase):
+    nested_objects = {}
+
     def __getattr__(self, name):
         try:
-            return super().__getitem__(name)
+            value = super().__getitem__(name)
         except KeyError:
-            return super().__getattr__(name)
+            value = super().__getattr__(name)
+
+        if name in self.nested_objects:
+            cls = getattr(sys.modules["welkin.models"], self.nested_objects[name])
+            if issubclass(cls, Collection):
+                obj = cls()
+                obj.extend([cls.resource(item) for item in value])
+                value = obj
```

Comment:
These three lines can be simplified to:

```python
value = cls(cls.resource(item) for item in value)
```

Since collection is a subclass of list, this is the same as passing a generator to a list constructor like:

```python
list(i for i in range(10))
```

Reply:
there's the magic we were tried